src/test/java/com/influxdb/v3/client/integration/E2ETest.java

```diff
@@ -0,0 +1,210 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.influxdb.v3.client.integration;
+
+import java.math.BigInteger;
+import java.net.ConnectException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.time.Instant;
+import java.util.Map;
+import java.util.UUID;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+import com.influxdb.v3.client.InfluxDBClient;
+import com.influxdb.v3.client.Point;
+import com.influxdb.v3.client.PointValues;
+import com.influxdb.v3.client.config.ClientConfig;
+import com.influxdb.v3.client.write.WriteOptions;
+import com.influxdb.v3.client.write.WritePrecision;
+
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
+
+public class E2ETest {
+
+    private static final java.util.logging.Logger LOG = Logger.getLogger(E2ETest.class.getName());
+
+    @EnabledIfEnvironmentVariable(named = "TESTING_INFLUXDB_URL", matches = ".*")
+    @EnabledIfEnvironmentVariable(named = "TESTING_INFLUXDB_TOKEN", matches = ".*")
+    @EnabledIfEnvironmentVariable(named = "TESTING_INFLUXDB_DATABASE", matches = ".*")
+    @Test
+    void testQueryWithProxy() {
+        String proxyUrl = "http://localhost:10000";
+
+        try {
+            // Continue to run this test only if Envoy proxy is running on this address http://localhost:10000
+            String url = String.format("http://%s:%d", "localhost", 10000);
+            URLConnection hpCon = new URL(url).openConnection();
+            hpCon.connect();
+        } catch (Exception e) {
+            if (e instanceof ConnectException && e.getMessage().contains("Connection refused")) {
+                LOG.warning("Tests with proxy have been skipped because no proxy is running on " + proxyUrl);
+                assumeFalse(e.getMessage().contains("Connection refused"));
```

Comment:
I added this line so that this test will be marked _skipped_ by junit when the proxy is not running.  I noticed it was marked as passing.